tests/core/io/test_tcp_server.h

```diff
@@ -35,12 +35,21 @@
 #include "core/io/tcp_server.h"
 #include "tests/test_macros.h"
 
+#include <functional>
+
 namespace TestTCPServer {
 
 const int PORT = 12345;
 const IPAddress LOCALHOST("127.0.0.1");
 const uint32_t SLEEP_DURATION = 1000;
-const uint64_t MAX_WAIT_USEC = 100000;
+const uint64_t MAX_WAIT_USEC = 2000000;
```

Comment:
I raised this so we could also test the timeout functionality when testing the connection to a server that is not listening.

This should not affect the timing of other tests unless they fail, but we can maybe make this a parameter of `wait_for_condition` and only have the higher value in that specific test.